controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1396,9 +1398,17 @@ func (r *BareMetalHostReconciler) doServiceIfNeeded(prov provisioner.Provisioner
 			servicingData.ActualFirmwareSettings = hfs.Status.Settings
 			servicingData.TargetFirmwareSettings = hfs.Spec.Settings
 		}
+
```

Comment:
You need a separate block that checks hup.Spec.FirmwareUpdates, similarly to how hup.Spec.FirmwareSettings defines liveFirmwareSettingsAllowed.

Reply:
oh you are right, we have `firmwareSettings` and `firmwareUpdates` in the Spec for the HUP!
I will update